File: src/test/java/com/alibaba/ttl/perf/memoryleak/NoMemoryLeak_TransmittableThreadLocal_NoRemove.java
package com.alibaba.ttl.perf.memoryleak;

import com.alibaba.ttl.TransmittableThreadLocal;
import com.alibaba.ttl.perf.Utils;

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
public class NoMemoryLeak_TransmittableThreadLocal_NoRemove {

    private NoMemoryLeak_TransmittableThreadLocal_NoRemove() {
        throw new InstantiationError("Must not instantiate this class");
    }

    public static void main(String[] args) {
        long counter = 0;
        while (true) {
            TransmittableThreadLocal<String> threadLocal = new TransmittableThreadLocal<>();
            threadLocal.set(Utils.getRandomString());

            if (counter % 1000 == 0)
                System.out.printf("%05dK%n", counter / 1000);
            counter++;
        }
    }
}

File: ttl2-compatible/src/test/java/com/alibaba/perf/memoryleak/NoMemoryLeak_TransmittableThreadLocal_NoRemove.kt
@file:JvmName("NoMemoryLeak_TransmittableThreadLocal_NoRemove")

package com.alibaba.perf.memoryleak

import com.alibaba.ttl.TransmittableThreadLocal
import com.alibaba.perf.getRandomString

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
fun main() {
    var counter: Long = 0
    while (true) {
        val threadLocal = TransmittableThreadLocal<String>()
        threadLocal.set(getRandomString())

        if (counter % 1000 == 0L)
            System.out.printf("%05dK%n", counter / 1000)
        counter++
    }
}
